pkg/pidfile/pidfile.go

```diff
@@ -23,27 +23,22 @@ import (
 	"sync"
 
 	"github.com/cilium/cilium/pkg/cleanup"
-	"github.com/cilium/cilium/pkg/logging"
-	"github.com/cilium/cilium/pkg/logging/logfields"
-
-	"github.com/sirupsen/logrus"
 )
 
 var (
-	log        = logging.DefaultLogger.WithField(logfields.LogSubsys, "pidfile")
 	cleanUPSig = make(chan struct{})
 	cleanUPWg  = &sync.WaitGroup{}
 )
 
 // Remove deletes the pidfile at the specified path. This does not clean up
 // the corresponding process, so should only be used when it is known that the
 // PID contained in the file at the specified path is no longer running.
-func Remove(path string) {
-	scopedLog := log.WithField(logfields.PIDFile, path)
-	scopedLog.Debug("Removing pidfile")
+func Remove(path string) error {
```

Comment:
Please update the callers to check (and log) the resulting error.

Reply:
Done! There was however only one direct call site outside of `pidfile` (not counting pidfile tests which do `defer Remove()` which I didn't change):
https://github.com/cilium/cilium/blob/3bb7b590a39c34dcfa678576adb78a13415df5b9/daemon/health.go#L53-L57

Unfortunately, the most common path to `pidfile.Remove()` is triggered asynchronously from `pidfile.Clean()` where there would need additional synchronization to propagate the error :/

Reply:
Yeah, well I think the `Clean()` tends to be run during shutdown anyway so I don't think we need to be quite so worried. LGTM.